### PR TITLE
Proj0450 & Proj0451: Test project should publish nor package

### PR DESCRIPTION
**Pull request by Corniel**
Packabillity and Publishabillity should be handled differently for test projects in that sense that they should always be set to false.

### CODE REVIEW

File: projects/PackableTestProject/PackableTestProject.csproj [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (LauraXiulan) [inline projects/PackableTestProject/PackableTestProject.csproj:1]: This and the implicit project are both for Packable and Publishable. Can you either separate the two or make it clear in the project name.

File: specs/DotNetProjectFile.Analyzers.Specs/DotNetProjectFile.Analyzers.Specs.csproj [hunk 2]
```diff
@@ -5,6 +5,7 @@
   <PropertyGroup>
     <TargetFramework>net8.0</TargetFramework>
     <IsTestProject>true</IsTestProject>
+    <IsPackable>true</IsPackable>
```
Comment (LauraXiulan) [inline specs/DotNetProjectFile.Analyzers.Specs/DotNetProjectFile.Analyzers.Specs.csproj:8]: Why is this added here?

File: src/DotNetProjectFile.Analyzers/Rule.cs [hunk 3]
```diff
@@ -447,6 +447,26 @@ public static class Rule
         tags: ["Configuration", "confusion"],
         category: Category.Clarity);
 
+    public static DiagnosticDescriptor TestProjectShouldNotBePackable => New(
+        id: 0450,
+        title: "Test projects should not be packable",
+        message: "Set <IsPackable> to false.",
+        description:
+            "Test projects should only be responsible for running tests. Hence " +
+            "the should not be packable.",
```
Comment (LauraXiulan) [inline src/DotNetProjectFile.Analyzers/Rule.cs:456]:
```suggestion
            "they should not be packable.",
```

File: src/DotNetProjectFile.Analyzers/Rule.cs [hunk 4]
```diff
@@ -447,6 +447,26 @@ public static class Rule
...
+        tags: ["Configuration"],
+        category: Category.Bug);
+
+    public static DiagnosticDescriptor TestProjectShouldNotBePublishable => New(
+        id: 0451,
+        title: "Test projects should not be publishable",
+        message: "Set <IsPublishable> to false.",
+        description:
+            "Test projects should only be responsible for running tests. Hence " +
+            "the should not be publishable.",
```
Comment (LauraXiulan) [inline src/DotNetProjectFile.Analyzers/Rule.cs:466]:
```suggestion
            "they should not be publishable.",
```

**Review by LauraXiulan — CHANGES_REQUESTED** [review]

File: projects/PackableTestProject/PackableTestProject.csproj (hunk 1 above)
Comment (Corniel) [reply projects/PackableTestProject/PackableTestProject.csproj:1]: Suggestion for the name?

File: specs/DotNetProjectFile.Analyzers.Specs/DotNetProjectFile.Analyzers.Specs.csproj (hunk 2 above)
Comment (Corniel) [reply specs/DotNetProjectFile.Analyzers.Specs/DotNetProjectFile.Analyzers.Specs.csproj]: Oops.

File: projects/PackableTestProject/PackableTestProject.csproj (hunk 1 above)
Comment (LauraXiulan) [reply projects/PackableTestProject/PackableTestProject.csproj:1]: PackableAndPublishableTestProject

Comment (LauraXiulan) [reply projects/PackableTestProject/PackableTestProject.csproj:1]: Or without the 'And'

**Review by CptWesley — APPROVED** [review]
